### PR TITLE
feat:create dispatch

### PR DESCRIPTION
**Pull request by Talaspaev**

### CODE REVIEW

File: src/Components/Profile/MyPosts/myPosts.jsx [hunk 1]
```diff
@@ -5,16 +5,16 @@ import Post from './Post/post';
...
 
   const addPost = () => {
-    areaFunc();
+    dispatch({ type: 'ADD-POST' });
   };
 
   const handlerPost = () => {
     const text = refArea.current.value;
-    updateArea(text);
+    dispatch({ type: 'UPDATE-POST', NewText: text });
```
Comment (Dmitry-White) [inline src/Components/Profile/MyPosts/myPosts.jsx:17]: Consider passing your data under "payload" key in the action object.
`const action = ({type: 'UPDATE-POST', payload: text});`

Comment (Dmitry-White) [inline src/Components/Profile/MyPosts/myPosts.jsx:17]: Consider storing {type: 'UPDATE-POST', payload: text}} in a separate variable called "action".

File: src/Components/Profile/MyPosts/myPosts.jsx [hunk 2]
```diff
@@ -5,16 +5,16 @@ import Post from './Post/post';
 import style from './myPosts.module.css';
 
 const MyPosts = (props) => {
-  const { areaFunc, updateArea } = props.props;
+  const { dispatch } = props.props;
   const refArea = React.createRef();
 
   const addPost = () => {
```
Comment (Dmitry-White) [inline src/Components/Profile/MyPosts/myPosts.jsx:11]: Same as for handlerPost function.

File: src/redux/state.js [hunk 3]
```diff
@@ -52,13 +55,22 @@ const store = {
...
-  updateArea(NewText) {
+  _updateArea(NewText) {
     this._state.ProfilePages.textArea = NewText;
     this._renderFunc(this._state);
   },
-  subscribe(observer) {
-    this._renderFunc = observer;
+  dispatch(action) {
+    switch (action.type) {
+      case 'ADD-POST':
```
Comment (Dmitry-White) [inline src/redux/state.js:64]: Consider creating a file named "actionTypes.js" and storing there an object with all the action names.
`const ACTION_TYPES = { ..., ADD_POST: 'ADD_POST', ...};`
This way you can import this object everywhere you need and reference needed type like this
`switch(ACTION_TYPES.ADD_POST)`

File: src/Components/Profile/MyPosts/myPosts.jsx (hunk 1 above)
Comment (Talaspaev) [reply src/Components/Profile/MyPosts/myPosts.jsx:17]: Sure .I should have to added this object in variable

Comment (Talaspaev) [reply src/Components/Profile/MyPosts/myPosts.jsx:17]: same above

Comment (Talaspaev) [reply src/Components/Profile/MyPosts/myPosts.jsx:17]: same above

File: src/redux/state.js (hunk 3 above)
Comment (Talaspaev) [reply src/redux/state.js]: I added new file and I used it in state and mypost

File: src/Components/Profile/MyPosts/myPosts.jsx (hunk 2 above)
Comment (Talaspaev) [reply src/Components/Profile/MyPosts/myPosts.jsx:12]: Same above

File: src/Components/Profile/MyPosts/myPosts.jsx [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
-    areaFunc();
+    const action = { type: ACTION_TYPES.ADD_POST };
+    dispatch(action);
   };
 
   const handlerPost = () => {
     const text = refArea.current.value;
-    updateArea(text);
+    const action = { type: ACTION_TYPES.UPDATE_POST, NewText: text };
```
Comment (Dmitry-White) [inline src/Components/Profile/MyPosts/myPosts.jsx:19]: Consider making an "action" object the same structure throughout your app - keep the data under "payload" key, not under a custom key like "NewText"

Comment (Talaspaev) [reply src/Components/Profile/MyPosts/myPosts.jsx]: I added payload in action

**Pull request merged**